src/ui/common/src/components/integrations/addIntegrations.tsx

```diff
@@ -54,96 +53,128 @@ const AddIntegrations: React.FC<Props> = ({
         {Object.entries(supportedIntegrations)
           .filter(([svc]) => svc !== 'Aqueduct Demo')
           .map(([svc, integration]) => {
-            if (integration.category !== category) {
-              return null;
-            }
-
-            const service = svc as Service;
-            const [showDialog, setShowDialog] = useState(false);
```

Comment:
Because of this useState call, the linter complained that we're not using hooks properly, as they should be called at the top level of a component, so I had to refactor this list and turn the map contents into a react component.